Rubberduck.RegexAssistant/Atom.cs

```diff
@@ -0,0 +1,284 @@
+﻿using Rubberduck.RegexAssistant.i18n;
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Rubberduck.RegexAssistant
+{
+    public interface Atom : IDescribable
+    {
+        string Specifier { get; }
+    }
+
+    public class CharacterClass : Atom
+    {
+        public static readonly string Pattern = @"(?<!\\)\[(?<expression>.*?)(?<!\\)\]";
+        private static readonly Regex Matcher = new Regex("^" + Pattern + "$");
+
+        public bool InverseMatching { get; }
```

Comment:
or

    public bool InverseMatching { get; private set; }

But I like the `private readonly` backing field more, because it says "this is an immutable property" and the `private set` one really says "this is a property that can only be set by the internal implementation"... which is quite different.